# Phase Rotation
### *Chapter 10 - Polyphase AC Circuits*

## Three-Phase Alternator

Let’s take the three-phase alternator design laid out earlier and watch what happens as the magnet rotates.

![three-phase-alternator](https://www.allaboutcircuits.com/uploads/articles/three-phase-alternator.jpg)

The phase angle shift of 120° is a function of the actual rotational angle shift of the three pairs of windings.

If the magnet is rotating clockwise, winding 3 will generate its peak instantaneous voltage exactly 120° (of alternator shaft rotation) after winding 2, which will hit its peak 120° after winding 1. The magnet passes by each pole pair at different positions in the rotational movement of the shaft.

Where we decide to place the windings will dictate the amount of phase shift between the windings’ AC voltage waveforms.

If we make winding 1 our “reference” voltage source for phase angle (0°), then winding 2 will have a phase angle of -120° (120° lagging, or 240° leading) and winding 3 an angle of -240° (or 120° leading).

## Phase Sequence

This sequence of phase shifts has a definite order. For clockwise rotation of the shaft, the order is 1-2-3 (winding 1 peak first, them winding 2, then winding 3). This order keeps repeating itself as long as we continue to rotate the alternator’s shaft.

![phase-sequence](https://www.allaboutcircuits.com/uploads/articles/clockwise-rotation-phase.jpg)

*Note that the above notion*, and anything that depends on it, including any plot, assumes that the positions of $2$b and $2$a are swapped in the above diagram and that positions of winding pairs $2$ and $3$ are subsequently swapped.  Without this assumption, winding $2$ would have a phase angle of $-60^\circ$ (i.e. it would lag by $60^\circ$) and winding $3$ would have a phase angle of $-120^\circ$ (i.e. it would lag by $120^\circ$).

Below is a diagram of an alternator with the correct winding configuration.  

![corrected-alternator](three-phase-alternator.svg)

Now, getting back to describing the phase sequence, when the rotation is reversed -- i.e., when the alternator's shaft is rotated in the counter-clockwise direction -- the phase sequence's order reverses accordingly.

![phase-sequence-reversed](https://www.allaboutcircuits.com/uploads/articles/counterclockwise-rotation-phase.jpg)

## Phase Sequence Detectors

Since voltmeters and ammeters would be useless in telling us what the phase rotation of an operating power system is, we need to have some other kind of instrument capable of doing the job.

One ingenious circuit design uses a capacitor to introduce a phase shift between voltage and current, which is then used to detect the sequence by way of comparison between the brightness of two indicator lamps in the figure below.

![phase-sequence-detector](https://www.allaboutcircuits.com/uploads/articles/phase-sequence-detector.jpg)

The two lamps are of equal filament resistance and wattage. The capacitor is sized to have approximately the same amount of reactance at system frequency as each lamp’s resistance.

If the capacitor were to be replaced by a resistor of equal value to the lamps’ resistance, the two lamps would glow at equal brightness, the circuit is balanced. However, the capacitor introduces a phase shift between voltage and current in the third leg of the circuit equal to 90°.

This phase shift, greater than 0° but less than 120°, skews the voltage and current values across the two lamps according to their phase shifts relative to phase 3.

### With Resistor

Herein is a full analysis with a resistor with the same resistance, and subsequently power dissapation or wattage, as the two lamps' filament resistance and wattage in place of capacitor connected to phase \#3.

Suppose all three loads have a resistance of $2650\,\Omega$ and are connected to a power supply (an alternator) with a voltage of $120 \mathrm{V}$ and phase angles, from \#$1$ to \#$3$, denoted as '$\theta_1$', '$\theta_2$', and '$\theta_3$' respectively, of $\theta_1 = 0$, $\theta_2 = 120^\circ$, and $\theta_3 = 240^\circ$.

The amplitude of each current is given by, via Ohm's Law,  

$$ |E| = R|I|$$

$$ |I| = \frac{|E|}{R} = \frac{120\mathrm{V}}{2650\,\Omega}\approxeq 45.283\,\mathrm{mA}$$

In [1]:
1000*120/2650

45.283018867924525

And the phase angle of each current, since the loads are balanced, is equal to the phase angle of the the voltage of it's respective load.

Now, let's **verify** this against KCL.  Via KCL,  

$$ I_1 + I_2 + I_3 = 0$$

$$ 45.283\,\mathrm{mA}\,\angle\,0 \:+ \:45.283\,\mathrm{mA}\,\angle\,120^\circ = - 45.283\,\mathrm{mA}\,\angle\,240^\circ = 45.283\,\mathrm{mA}\,\angle\,60^\circ$$

In rectangular form, the above can be rewritten as:  

$$ 45.283\,\mathrm{mA}\,\big(1 + \cos120^\circ + j\cdot\sin120^\circ \big) = 45.283\,\mathrm{mA}\big(\cos60^\circ + j\cdot\sin60^\circ \big)$$

Dividing throughout by the amplitude gives:  

$$ \big(1 + \cos120^\circ + j\cdot\sin120^\circ\big ) = \big(\cos60^\circ + j\cdot\sin60^\circ\big)$$

In [2]:
import numpy as np

In [3]:
(np.cos(np.radians(120)), np.sin(np.radians(120)))

(-0.4999999999999998, 0.8660254037844387)

In [4]:
(np.cos(np.radians(60)), np.sin(np.radians(60)))

(0.5000000000000001, 0.8660254037844386)

Note that, throughout this subsection, it is assumed that the polarity of the power system is the same as the one shown below, in the next subsection, with the added assumption that the positive poles of each load are adjacent to the positive poles of their respective power supply.

### With Capacitor

Suppose we have the following circuit:  

![phase-sequence-detector-w/capacitor](https://www.allaboutcircuits.com/uploads/articles/spice-circuit-for-phase-sequence-detector.jpg)

First, let's consider that the frequency ($f$) can be found via the presuppostion that the reactance of $\mathrm{C}_1$, denoted as '$X_{\mathrm{C}1}$', is equal to $2650\,\Omega$.  

$$ X_{\mathrm{C}1} = \frac{1}{2\pi f C_1} = 2650\,\Omega = R_1, R_2$$

$$ f = \frac{1}{2\pi \cdot 2650\,\Omega\cdot1\mu\mathrm{F}}\approxeq60\mathrm{Hz}$$

In [5]:
f = 1 / (2*np.pi*2650*1e-6)
f

60.05846909128126

Assuming that the positive poles of each load are adjacent to the positive poles of their respective power supplies, via KVL and Ohm's Law,  

\begin{equation}\tag{1}
    120\mathrm{V}\angle\,0\:-\:R_1I_1 \:-\:(I_1+I_2)\,R_1\angle\!-90^\circ\:-\:120\mathrm{V}\angle\,240^\circ = 0
\end{equation}    

and,
\begin{equation}\tag{2}
    120\mathrm{V}\angle\,120^\circ = R_1I_2 \:+\: (I_1 +I_2)\,R_1\angle\!-90^\circ \:+\: 120\mathrm{V}\angle\,240^\circ
\end{equation}    

\begin{equation}\tag{2a}
    I_1 R_1\angle\!-90^\circ =  120\mathrm{V}\angle\,120^\circ \:-\: R_1I_2 \:-\: I_2R_1\angle\!-90^\circ \:-\: 120\mathrm{V}\angle\,240^\circ
\end{equation}    

Dividing throughout by $1\angle\!-90^\circ$ gives:  

\begin{equation}\tag{2b}
    R_1I_1 = 120\mathrm{V}\angle\,210^\circ \:-\:R_1I_2\angle\,90^\circ \:-\:R_1I_2 \:-\: 120\mathrm{V}\angle\,330^\circ
\end{equation}    

Via equations (2a), (2b), and (1) in conjunction with substitution,  

$$ 120\mathrm{V}\angle\,0 = 120\mathrm{V}\angle\,210^\circ \:-\:R_1I_2\angle\,90^\circ \:-\:R_1I_2 \:-\: 120\mathrm{V}\angle\,330^\circ \:+\: R_1I_2\angle\!-90^\circ \:+\: 120\mathrm{V}\angle\,120^\circ \:-\: R_1I_2 \:-\: I_2R_1\angle\!-90^\circ \:-\: 120\mathrm{V}\angle\,240^\circ + 120\mathrm{V}\angle\,240^\circ$$

We can solve for $I_2$ in the above equation, but first let's simplify it.

\begin{align*}\tag{1a}
    2R_1I_2 + R_1I_2\angle\,90^\circ &= 120\mathrm{V}\big(1\angle\,210^\circ - 1 - 1\angle\,330^\circ + 1\angle\,120^\circ \big) \\
    R_1I_2(2 + 1\angle\,90^\circ)&= 120\mathrm{V}\big(1\angle\,210^\circ - 1 - 1\angle\,330^\circ + 1\angle\,120^\circ \big) \\
    I_2 &= \frac{120\mathrm{V}}{R_1(2 + 1\angle\,90^\circ)}\,\big(1\angle\,210^\circ - 1 - 1\angle\,330^\circ + 1\angle\,120^\circ \big)
\end{align*}    

To convert $I_2$ to rectangular form, let's break it's identity down into it's numerator and denominator.  The numerator, denoted as '$\mathrm{num}$', in rectangular form is given by:  

\begin{align*}
    \mathrm{num} &= 120\mathrm{V}\,\big(1\angle\,210^\circ - 1 - 1\angle\,330^\circ + 1\angle\,120^\circ \big) \\
     &= 120\mathrm{V}\,\big(\cos210^\circ - 1 - \cos330^\circ + \cos120^\circ + j\cdot(\sin210^\circ - \sin330^\circ + \sin120^\circ) \big) \\
     &\approxeq 120\mathrm{V}\,(-3.23 + j\cdot 0.866)
\end{align*}     

In [6]:
def cos(deg: float):
    return np.cos(np.radians(deg))

def sin(deg: float):
    return np.sin(np.radians(deg))

In [7]:
num = complex(cos(210) - 1 - cos(330) + cos(120), sin(210) -sin(330) + sin(120))
num

(-3.2320508075688767+0.866025403784439j)

And the denominator ($\mathrm{den}$) is given by:  

\begin{align*}
    \mathrm{den} &= R_1(2 + 1\angle\,90^\circ) \\
     &= R_1(2 + j)
\end{align*}

In [8]:
den = complex(2, 1)
num/den

(-1.119615242270663+0.9928203230275511j)

Via the above two equation, equation (1a), and substitution,  

$$ I_2 =\frac{\mathrm{num}}{\mathrm{den}} \approxeq \frac{120\mathrm{V}}{2650\,\Omega}\,(-1.12 + j)$$

In [9]:
np.sqrt(1.12**2+1)*120/2650

0.06799091095833405

In [10]:
np.degrees(np.arctan(-1/1.12))

-41.76029970389787

In polar form:  

\begin{equation}\tag{1a}
    I_2\approxeq 67.99\,\mathrm{mA}\,\angle\!-41.76^\circ
\end{equation}    

We can now solve for $I_1$ in equation (2).  Via the above equation, equation (2), and substitution,  

$$ 120\mathrm{V}\angle\,120^\circ \approxeq R_1\cdot67.99\,\mathrm{mA}\,\angle\!-41.76^\circ \:+\: R_1I_1\,\angle\!-90^\circ \:+\: R_1\cdot67.99\,\mathrm{mA}\,\angle\!-131.76^\circ \:+\:120\mathrm{V}\angle\,240^\circ$$

$$ R_1I_1\,\angle\!-90^\circ \approxeq 120\mathrm{V}\angle\,120^\circ \:-\: R_1\cdot67.99\,\mathrm{mA}\,\angle\!-41.76^\circ \:-\: R_1\cdot67.99\,\mathrm{mA}\,\angle\!-131.76^\circ \:-\:120\mathrm{V}\angle\,240^\circ$$

Dividing throughout by $R_1\,\angle\!-90^\circ$ gives:  

\begin{align*}\tag{1b}
    I_1 &\approxeq \frac{120\mathrm{V}}{2650\,\Omega}\,\angle\,210^\circ \:-\: 67.99\,\mathrm{mA}\,\angle\,48.24^\circ \:-\: 67.99\,\mathrm{mA}\,\angle\!-41.76^\circ \:-\: \frac{120\mathrm{V}}{2650\,\Omega}\,\angle\,330^\circ \\ \\
     &\approxeq 45.28\,\mathrm{mA}\,\angle\,210^\circ \:-\: 67.99\,\mathrm{mA}\,\angle\,48.24^\circ \:-\: 67.99\,\mathrm{mA}\,\angle\!-41.76^\circ \:-\: 45.28\,\mathrm{mA}\,\angle\,330^\circ \\
     &\approxeq -(174.43 + j\cdot5.43)\,\mathrm{mA}
\end{align*}

In [14]:
120/2650

0.045283018867924525

In [15]:
I_1 = complex(45.28*cos(210) - 67.99*cos(48.24) - 67.99*cos(-41.76) - 45.28*cos(330), \
             45.28*sin(210) - 67.99*sin(48.24) - 67.99*sin(-41.76) - 45.28*sin(330))

I_1

(-174.42594591096366-5.434391610535364j)

Finally, in polar form:  

\begin{equation}\tag{1b}
     I_1\approxeq174.51\,\mathrm{mA}\,\angle\,181.78^\circ
\end{equation}

In [19]:
np.sqrt(I_1.real**2 + I_1.imag**2)

174.51058196886248

In [21]:
np.degrees(np.arctan(5.43/174.43)) + 180

181.7830398606807

The **voltage** running across load \#$1$, denoted as '$E_1$', is, via Ohm's Law, equation (1b), and substitution,  

$$ E_1 = R_1I_1 \approxeq 2650\,\Omega\cdot174.51\,\mathrm{mA}\,\angle\,181.78^\circ \approxeq 462.45\mathrm{V}\angle181.78^\circ$$

In [22]:
2.65*174.51

462.45149999999995